### PR TITLE
Implement recenter button

### LINKED ISSUE #3946: Recenter button
@LukasPaczos commented on [Tue Jan 26 2021](https://github.com/mapbox/mapbox-navigation-android-internal/issues/94)

Tracks the re-implementation of the sound control button available in the v1 into v2 - https://github.com/mapbox/mapbox-navigation-android/blob/master/libnavigation-ui/src/main/java/com/mapbox/navigation/ui/RecenterButton.java.

We should follow the new API-View architecture and add appropriate unit and instrumentation tests.

@d-prukop are you considering any new designs for the recenter button?

/cc @abhishek1508

### PR DESCRIPTION
**Pull request by abhishek1508**
### Description

Fixes #3946 

### Changelog

```
<changelog>Implementation of recenter button.</changelog>
```

### CODE REVIEW

**Comment by RingerJK** [comment]
Animation is not running as expected and part of the text is hidden. The device is `Samsung S8`(SM-G950U), Android `7.0`
https://user-images.githubusercontent.com/8738058/113638624-f330b280-967f-11eb-91a5-c5ca593f24df.mp4

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/view/MapboxRecenterButton.kt [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                    binding.recenterText.text = context.getString(R.string.mapbox_recenter)
+                    binding.recenterText.visibility = View.VISIBLE
+                    mainHandler.postDelayed(
+                        {
+                            val endAnimator = getAnimator(extendToWidth.toInt(), textWidth)
+                            binding.recenterText.shrink(
+                                endAnimator,
+                                doOnStart = {
+                                    binding.recenterText.text = null
+                                    isAnimationRunning = false
```
Comment (RingerJK) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/view/MapboxRecenterButton.kt:106]: `isAnimationRunning = false` should be moved under `doOnEnd`. Here and below(`MapboxRouteOverviewButton.kt `)

**Comment by Guardiola31337** [comment]
Took this for a quick spin and it's looking good so far.

Could you fix [`static-analysis`](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-android/11257/workflows/4cd7a33e-4a5f-4bcd-bc58-297ac8741cee/jobs/44865) when you have a chance? `ktlint` is ❌ 

```
> Task :libnavui-maps:ktlint
/root/code/libnavui-maps/src/test/java/com/******/navigation/ui/maps/camera/view/MapboxRecenterButtonTest.kt:4:1: Unused import
/root/code/libnavui-maps/src/test/java/com/******/navigation/ui/maps/camera/view/MapboxRecenterButtonTest.kt:10:1: Unused import
/root/code/libnavui-maps/src/test/java/com/******/navigation/ui/maps/camera/view/MapboxRecenterButtonTest.kt:11:1: Wildcard import (cannot be auto-corrected)
/root/code/libnavui-maps/src/test/java/com/******/navigation/ui/maps/camera/view/MapboxRouteOverviewButtonTest.kt:4:1: Unused import
/root/code/libnavui-maps/src/test/java/com/******/navigation/ui/maps/camera/view/MapboxRouteOverviewButtonTest.kt:9:1: Unused import
/root/code/libnavui-maps/src/test/java/com/******/navigation/ui/maps/camera/view/MapboxRouteOverviewButtonTest.kt:10:1: Unused import
/root/code/libnavui-maps/src/test/java/com/******/navigation/ui/maps/camera/view/MapboxRouteOverviewButtonTest.kt:12:1: Unused import
/root/code/libnavui-maps/src/test/java/com/******/navigation/ui/maps/camera/view/MapboxRouteOverviewButtonTest.kt:13:1: Wildcard import (cannot be auto-corrected)

> Task :libnavui-maps:ktlint FAILED
```

File: libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/view/MapboxSoundButton.kt [hunk 2]
```diff
@@ -215,7 +215,7 @@ class MapboxSoundButton : ConstraintLayout {
 
     private companion object {
         const val SLIDE_DURATION = 300L
-        const val EXTEND_MUTED_TO_WIDTH = 140
+        const val EXTEND_MUTED_TO_WIDTH = 150
```
Comment (Guardiola31337) [inline libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/view/MapboxSoundButton.kt:218]: Should this be part of https://github.com/mapbox/mapbox-navigation-android/pull/4219?

File: libnavui-voice/src/main/res/layout/mapbox_sound_button_layout.xml [hunk 3]
```diff
@@ -27,8 +27,7 @@
         android:layout_width="@dimen/mapbox_dimen_56dp"
         android:layout_height="@dimen/mapbox_dimen_56dp"
         app:elevation="0dp"
-        android:layout_margin="@dimen/mapbox_dimen_2dp"
-        android:padding="@dimen/mapbox_dimen_18dp"
+        android:padding="@dimen/mapbox_dimen_16dp"
```
Comment (Guardiola31337) [inline libnavui-voice/src/main/res/layout/mapbox_sound_button_layout.xml:30]: Same as above. Should this be part of https://github.com/mapbox/mapbox-navigation-android/pull/4219?

**Review by Guardiola31337 — APPROVED** [review]

**Pull request merged**